File: 0062-unique-paths/0062-unique-paths.c
long long uniquePaths(int m, int n) {
    int N = m + n - 2;  // Total number of steps the robot needs to take
    int k = m - 1;  // Choose the smaller of (m-1) and (n-1) to optimize the calculation
    if (k > N - k) k = N - k;  // Since C(N, k) == C(N, N-k)
    
    long long res = 1;
    
    for (int i = 1; i <= k; i++) {
        res = res * (N - i + 1) / i;
    }
    
    return res;
}

File: 0062-unique-paths/0062-unique-paths.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[][] dp = new int[m][n];
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(i == 0 || j == 0){
                    dp[i][j] = 1;
                }else {
                    dp[i][j] = dp[i][j-1] + dp[i-1][j];
                }
            }
        }
        return dp[m-1][n-1];
        // if( n == 1 || m == 1) return 1;
        // else{
        //     return uniquePaths(m-1, n) + uniquePaths(m, n-1);
        // }
    }
}